SWLOR.Game.Server/Placeable/ControlTower/OnDamaged.cs

```diff
@@ -50,10 +53,20 @@ public bool Run(params object[] args)
             PCBaseStructure structure = _data.Single<PCBaseStructure>(x => x.ID == new Guid(structureID));
             int maxShieldHP = _base.CalculateMaxShieldHP(structure);
             PCBase pcBase = _data.Get<PCBase>(structure.PCBaseID);
+            var playerIDs = _data.Where<PCBasePermission>(x => x.PCBaseID == structure.PCBaseID).Select(s => s.PlayerID);
+            var toNotify = NWModule.Get().Players.Where(x => playerIDs.Contains(x.GlobalID));
+            bool alerted = false;
+            if (alerted == false)
+            {
+                foreach(NWPlayer player in toNotify)
+                {
+                    player.SendMessage("One of your bases is under attack!");
```

Comment:
I would add the area's name and the sector it's in to this message. Something like:

"Your base in <Area Name> [<Sector>] is under attack!"

Example: Your base in Viscara Wildlands [SW] is under attack!

There are helper methods in the BaseService which should get the sector for you. You'll have to pull back the PCBase from the data service to get the area resref. Then you can either pull from the Area DB entry or just pull it back with NWModule.Get().Areas.Single(x => x.Resref == pcBase.AreaResref)  - Whatever's easiest for you.